src/pubd/repository.rs

```diff
@@ -1363,17 +1377,81 @@ impl PublisherStats {
         self.size
     }
 
+    /// Returns the most recent "this_update" time
+    /// from all manifest(s) published by this publisher,
+    /// if any.. i.e. there may be 0, 1 or many manifests
     pub fn last_update(&self) -> Option<Time> {
-        self.last_update
+        let mut last_update = None;
+        for mft in self.manifests() {
+            if let Some(seen_update) = last_update {
+                let this_update = mft.this_update();
+                if this_update > seen_update {
+                    last_update = Some(this_update)
+                }
+            } else {
+                last_update = Some(mft.this_update());
+            }
+        }
+
+        last_update
+    }
+
+    pub fn manifests(&self) -> &Vec<PublisherManifestStats> {
+        &self.manifests
     }
 }
 
 impl From<&CurrentObjects> for PublisherStats {
     fn from(objects: &CurrentObjects) -> Self {
+        let mut manifests = vec![];
+        for el in objects.elements() {
+            if el.uri().ends_with("mft") {
+                if let Ok(mft) = Manifest::decode(el.base64().to_bytes().as_ref(), false) {
+                    if let Ok(stats) = PublisherManifestStats::try_from(&mft) {
+                        manifests.push(stats)
+                    }
+                }
+            }
+        }
+
         PublisherStats {
             objects: objects.len(),
             size: objects.size(),
-            last_update: None,
+            manifests,
         }
     }
 }
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PublisherManifestStats {
+    uri: uri::Rsync,
+    this_update: Time,
+    next_update: Time,
+}
+
+impl PublisherManifestStats {
+    pub fn uri(&self) -> &uri::Rsync {
+        &self.uri
+    }
+
+    pub fn this_update(&self) -> Time {
+        self.this_update
+    }
+
+    pub fn next_update(&self) -> Time {
+        self.next_update
+    }
+}
+
+impl TryFrom<&Manifest> for PublisherManifestStats {
+    type Error = ();
+
+    fn try_from(mft: &Manifest) -> Result<Self, Self::Error> {
+        let uri = mft.cert().signed_object().cloned().ok_or(())?;
```

Comment:
What will `uri` be here if this fails?

Reply:
An `Err(())`. In this case treat this manifest as syntactically incorrect (which it is - I added a comment as well now) - and we do not include it in our stats about this publisher.

Note that other API end-points let operators review the complete set of objects for publishers. This is just a quick overview. Also, incorrect objects should not be found in practice - as Krill and other conformant CAs will include this.